src/components/search-box/voice-search/VoiceSearchDialog.vue

```diff
@@ -0,0 +1,263 @@
+<script lang="ts" setup>
+import { ref, computed, watch } from 'vue'
+import { VoiceSearchOptions } from '@/types/search-box/SearchBoxOptions';
+import { getVoiceServiceApiUrl } from '@/utils/api.utils';
+
+const socket = ref<WebSocket | null>(null)
+
+const isRecordingRef = ref<boolean>(false)
+const mediaStream = ref<MediaStream | null>(null)
+const mediaRecorder = ref<MediaRecorder | null>(null)
+const timesliceLimit = ref<number>(4)
+const timeSliceLength = 1000
+const transcription = ref('')
+const stopDelay = 700
+const progressBar = ref<HTMLElement | null>(null)
+
+const props = defineProps<{
+  isOpen: boolean,
+  options: VoiceSearchOptions
+}>()
+
+const emit = defineEmits([
+  'close',
+  'transcript-update',
+  'stop-recognize'
+])
+
+watch(transcription, (newValue) => {
+  emit('transcript-update', newValue)
+})
+
+const description = computed(() => {
+  if (!isRecordingRef.value) {
+    return 'Microphone is off. Try again.'
```

Comment:
If this text is shown to the user, it should be configurable label within voice search options

Reply:
Same as with `Listnening` afterwards